### PR TITLE
Working basecase, dependent on ObjectEntityService refactor

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: api/src/ActionHandler/ZaakTypeHandler.php [hunk 1]
```diff
@@ -42,6 +42,8 @@ public function __run(array $data, array $configuration): array
         $entity = $this->entityManager->getRepository('App:Entity')->findOneBy(['id' => $configuration['entityId']]);
         $objectEntities = $this->entityManager->getRepository('App:ObjectEntity')->findByEntity($entity, ['identificatie' => $identifier]);
 
+        var_dump(count($objectEntities));
```
Comment (smisidjan) [inline api/src/ActionHandler/ZaakTypeHandler.php:45]: Here you forgot to delete a var_dump

Comment (rjzondervan) [reply api/src/ActionHandler/ZaakTypeHandler.php]: Well spotted, thanks! Fixed it now :)

**Review by smisidjan — APPROVED** [review]

File: api/src/Service/SynchronizationService.php [hunk 2]
```diff
@@ -234,50 +228,36 @@ private function findSyncByObject(ObjectEntity $objectEntity, Gateway $source, E
...
     {
         // We need an object on the gateway side
-        if (!$sync->getObject()){
+        if (!$sync->getObject()) {
             $object = new ObjectEntity();
             $object->setEntity($sync->getEntity());
             $sync->setObject($object);
-
+            $this->entityManager->persist($object);
             // todo: set organization and application for $object, see eavService->getObject() function.
```
Comment (WilcoLouwerse) [inline api/src/Service/SynchronizationService.php:236]: Are you sure this works without setting the organization and application (from the session, see eavService->getObject()) ?

File: api/src/Service/SynchronizationService.php [hunk 3]
```diff
@@ -234,50 +228,36 @@ private function findSyncByObject(ObjectEntity $objectEntity, Gateway $source, E
...
         $hash = hash('sha384', serialize($sourceObject));
-        // Lets turn the source into a dot so that we can grab values
         $dot = new Dot($sourceObject);
 
-        // Now we need to establish the last time the source was changed
-        if (array_key_exists('modifiedDateLocation', $configuration)) {
-            $lastChanged = $dot->get($configuration['modifiedDateLocation']);
-            $sync->setSourcelastChanged($lastChanged);
-        }
-        // What if the source has no property that allows us to determine the last change
```
Comment (WilcoLouwerse) [inline api/src/Service/SynchronizationService.php:260]: Removing to many of these comments will make the code less readable and understandable, would not recommend doing this, even if the result is less lines of text/code...

File: api/src/Service/SynchronizationService.php [hunk 4]
```diff
@@ -290,6 +270,29 @@ private function syncToSource(Synchronization $sync): Synchronization
         return $sync;
     }
 
+    private function populateObject(array $data, ObjectEntity $objectEntity): ObjectEntity
+    {
+        $owner = $this->objectEntityService->checkAndUnsetOwner($data);
+
+        if ($validationErrors = $this->validatorService->validateData($data, $objectEntity->getEntity(), 'POST')) {
```
Comment (WilcoLouwerse) [inline api/src/Service/SynchronizationService.php:277]: What are we going to do with validationErrors?

File: api/src/Service/SynchronizationService.php (hunk 2 above)
Comment (WilcoLouwerse) [reply api/src/Service/SynchronizationService.php:236]: I think even the saveObject route from the objectEntityService might not do this. Are we sure the org. and appl. of the main object (not subObjects!) are set before we persist and flush this object?

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: api/src/Service/SynchronizationService.php (hunk 3 above)
Comment (rjzondervan) [reply api/src/Service/SynchronizationService.php:260]: Partially agree, sometimes a comment is necessary, but if the internals of the code are obvious, adding comments about it only decreases the readability of the code.

File: api/src/Service/SynchronizationService.php (hunk 4 above)
Comment (rjzondervan) [reply api/src/Service/SynchronizationService.php:278]: I left that open for now indeed, it is something we need to look into, but I recommend putting it in logs.

Comment (WilcoLouwerse) [reply api/src/Service/SynchronizationService.php:329]: add a nice todo for this?

File: api/src/Service/SynchronizationService.php [hunk 5]
```diff
@@ -293,7 +354,38 @@ private function translateInput($configuration, $externObject): array
...
+    /**
+     * Sets an application and organization for new ObjectEntities.
+     *
+     * @param ObjectEntity $objectEntity The ObjectEntity to update
+     *
+     * @return ObjectEntity The updated ObjectEntity
+     */
+    private function setApplicationAndOrganization(ObjectEntity $objectEntity): ObjectEntity
+    {
+        $application = $this->entityManager->getRepository('App:Application')->findOneBy(['name' => 'main application']);
```
Comment (WilcoLouwerse) [inline api/src/Service/SynchronizationService.php:367]: Else, if only one application exists, use that one

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**